core/src/main/scala/quasar/datasource/kafka/KafkaDatasourceModule.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.datasource.kafka
+
+import slamdata.Predef._
+
+import argonaut.Argonaut._
+import argonaut._
+import cats.effect.{ConcurrentEffect, ContextShift, Resource, Timer}
+import cats.implicits._
+import cats.kernel.Hash
+import eu.timepit.refined.auto._
+import quasar.RateLimiting
+import quasar.api.datasource.DatasourceError.InitializationError
+import quasar.api.datasource.{DatasourceError, DatasourceType}
+import quasar.connector.datasource.LightweightDatasourceModule.DS
+import quasar.connector.datasource.{LightweightDatasourceModule, Reconfiguration}
+import quasar.connector.{ByteStore, MonadResourceErr}
+import scalaz.NonEmptyList
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Either, Left, Right}
+
+object KafkaDatasourceModule extends LightweightDatasourceModule {
+  override def kind: DatasourceType = DatasourceType("kafka", 1L)
+
+  override def sanitizeConfig(config: Json): Json = config.as[Config].result match {
+    case Left(_) => config
+    case Right(cfg) => cfg.sanitize.asJson
+  }
+
+  override def reconfigure(
+    original: Json,
+    patch: Json): Either[DatasourceError.ConfigurationError[Json], (Reconfiguration, Json)] = {
+    val back = for {
+      original <- original.as[Config].result.leftMap(_ =>
+        DatasourceError
+          .MalformedConfiguration[Json](
+            kind,
+            sanitizeConfig(original),
+            "Source configuration in reconfiguration is malformed."))
+
+      patch <- patch.as[Config].result.leftMap(_ =>
+        DatasourceError
+          .MalformedConfiguration[Json](
+            kind,
+            sanitizeConfig(patch),
+            "Patch configuration in reconfiguration is malformed."))
+
+      reconfigured <- original.reconfigure(patch).leftMap(c =>
```

Comment:
Could you rename `patch` and `original` so that they are not shadowing the values of the same name in the function definition? Imo, it's confusing to read (I was just looking all over for a function called `reconfigure` defined on a value of type `Json`, but here `original` is actually of type `Config`).

Reply:
Sure. This was essentially copy&paste, so I brought all the sins of the original with it. :)

Reply:
😄 I may well be the original sinner....